packages/kbn-es-query/src/filters/lib/saved_query_filter.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Filter, FilterMeta } from './meta_filter';
+import { RangeFilterParams } from './range_filter';
+
+interface Query {
+  query: string | { [key: string]: any };
+  language: string;
+}
+
+interface TimeRange {
+  timestamp: RangeFilterParams;
+  format: string;
+  from?: string;
+  to?: string;
+}
+
+export interface SavedQuery {
+  id: string;
+  attributes: SavedQueryAttributes;
+}
+
+interface EsQueryConfig {
+  allowLeadingWildcards: boolean;
+  queryStringOptions: { [key: string]: any };
+  dateFormatTZ: string | null;
+}
+interface IndexPattern {
+  id?: string;
+  title: string;
+  url?: string;
+  active?: boolean;
+  default?: boolean;
+  tag?: string[];
+  sort?: string;
+}
+
+interface SavedQueryAttributes {
+  title: string;
+  description: string;
+  query: Query;
+  filters?: Filter[];
+  timefilter?: SavedQueryTimeFilterConverted;
+}
+
+interface SavedQueryTimeFilterConverted {
+  range?: TimeRange;
+}
+
+export interface SavedQueryFilterParams {
+  savedQuery?: SavedQuery;
+  esQueryConfig: EsQueryConfig;
+  indexPattern: IndexPattern;
+}
+
+export type SavedQueryFilterMeta = FilterMeta & {
+  params: SavedQueryFilterParams;
+};
+
+export type SavedQueryFilter = Filter & {
```

Comment:
I'm thinking I need to move savedQuery, esConfig and index pattern out of params and into the filter itself.